devdoc/constbuffer_array_requirements.md

```diff
@@ -243,6 +243,58 @@ MOCKABLE_FUNCTION(, CONSTBUFFER_ARRAY_HANDLE, constbuffer_array_remove_front, CO
 
 **SRS_CONSTBUFFER_ARRAY_02_036: [** If there are any failures then `constbuffer_array_remove_front` shall fail and return `NULL`. **]**
 
+### constbuffer_array_add_back
+
+```c
+MOCKABLE_FUNCTION(, CONSTBUFFER_ARRAY_HANDLE, constbuffer_array_add_back, CONSTBUFFER_ARRAY_HANDLE, constbuffer_array_handle, CONSTBUFFER_HANDLE, constbuffer_handle);
+```
+
+`constbuffer_array_add_back` adds a new `CONSTBUFFER_HANDLE` at the back of the already stored `CONSTBUFFER_HANDLE`s.
+
+**SRS_CONSTBUFFER_ARRAY_05_001: [** If `constbuffer_array_handle` is `NULL` then `constbuffer_array_add_back` shall fail and return `NULL`.**]**
+
+**SRS_CONSTBUFFER_ARRAY_05_002: [** If `constbuffer_handle` is `NULL` then `constbuffer_array_add_back` shall fail and return `NULL`.**]**
+
+**SRS_CONSTBUFFER_ARRAY_05_003: [** `constbuffer_array_add_back` shall allocate enough memory to hold all of `constbuffer_array_handle` existing `CONSTBUFFER_HANDLE` and `constbuffer_handle`. **]**
+
+**SRS_CONSTBUFFER_ARRAY_05_004: [** `constbuffer_array_add_back` shall copy `constbuffer_handle` and all of `constbuffer_array_handle` existing `CONSTBUFFER_HANDLE`. **]**
+
+**SRS_CONSTBUFFER_ARRAY_05_005: [** `constbuffer_array_add_back` shall inc_ref all the `CONSTBUFFER_HANDLE` it had copied. **]**
+
+**SRS_CONSTBUFFER_ARRAY_05_006: [** `constbuffer_array_add_back shall` succeed and return a non-`NULL` value. ]**
```

Comment:
somewhere here it should say that it will build the function result by calling (probably) constbuffer_array_create

Reply:
Line 8 indicates that addition / deletion are immutable operation, so the result being created by calling constbuffer_array_create is implied?

I am fine either way.